Adds MoonTheBird's station objectives.

Aranclanos opened the pull request:
### Purpose
Adds MoonTheBird's station objectives.

Cargo manager will now save all its sent itemtraits in a history dictionary.
The station objective to send sheets will now check this history dictionary to see if the objective is met at the end of the round, removing the unityevent.
Removes the object description text away from ReportTemplates into the objective asset file itself.
Moves back asteroid spawning code to CentComm.cs
Changes the hardcoded string replacement for the StringBuilder functions.
Simplifies station objective code to be easier to follow and more straightforward.
Finishes #6081

Aranclanos committed:
Adds MoonTheBird's station objectives.

Cargo manager will now save all its sent itemtraits in a history dictionary.
The station objective to send sheets will now check this history dictionary to see if the objective is met at the end of the round, removing the unityevent.
Removes the object description text away from ReportTemplates into the objective asset file itself.
Moves back asteroid spawning code to CentComm.cs
Changes the hardcoded string replacement for the StringBuilder functions.
Simplifies station objective code to be easier to follow and more straightforward.

corp-0 commented on UnityProject/Assets/Scripts/Managers/CentComm.cs:
```diff
@@ -217,15 +214,18 @@ private void SpawnReports(string text)
 		/// </summary>
 		/// <param name="text">String that will be the report body</param>
 		/// <param name="type">Value from the UpdateSound enum to play as sound when announcing</param>
```
update the docstring to represent the param change

corp-0 commented on UnityProject/Assets/Scripts/Systems/StationObjectives/StationObjective.cs:
```diff
@@ -0,0 +1,45 @@
+		/// The description of the objective which is shown to players.
+		/// </summary>
+		[SerializeField]
+		public string description;
```
make it private with a public getter or remove the decorator since publics are serialzied by default (please choose former)

Aranclanos committed:
Removes the useless serializefield constructor from StationObjective description.
Corrects the params comments from the MakeCommandReport to its updated version.
